core/src/HIP/Kokkos_HIP_Abort.hpp

```diff
@@ -53,7 +53,7 @@
 namespace Kokkos {
 namespace Impl {
 
-[[noreturn]] __device__ __attribute__((noinline)) void hip_abort(
+[[noreturn]] __device__ __attribute__((noinline)) inline void hip_abort(
```

Comment:
Be kind to the reader and comment here.
```suggestion
// The two keywords below are not contradictory.  `noinline` is a directive to the optimizer.
[[noreturn]] __device__ __attribute__((noinline)) inline void hip_abort(
```

Reply:
Yes please.